### PR TITLE
Add docs for Sharp GP2Y1010AU0F PM2.5 sensor

### PR DESCRIPTION
**Pull request by zry98**
## Description:

**Related issue (if applicable):** N/A

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#6007

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/gp2y1010au0f.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+=======================================
+
+.. seo::
+    :description: Instructions for setting up Sharp GP2Y1010AU0F particulate matter sensor
+    :image: gp2y1010au0f.png
+
+The ``gp2y1010au0f`` sensor platform allows you to use your Sharp GP2Y1010AU0F particulate matter sensor with ESPHome.
+
+It is a very cheap (read: inaccurate) optical (infrared LED and phototransistor) sensor suitable for detecting fine particulate matter like dust and smoke. There are many much better sensors on the market and `supported by ESPHome </search.html?q=Particulate>`__, but this is one of the most common and cheapest ones you will find.
+
```
Comment (optimusprimespace) [inline components/sensor/gp2y1010au0f.rst:11]: You can include here the necessary components to have a functioning setup, 150k ohm resistor & 220 µF capacitor.

File: components/sensor/gp2y1010au0f.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+While the official `datasheet <https://global.sharp/products/device/lineup/data/pdf/datasheet/gp2y1010au_e.pdf>`__ doesn't mention its sensing resolution (whether it can detect PM2.5 particle or not), some vendors (e.g., `Waveshare <https://www.waveshare.com/wiki/Dust_Sensor>`__) say it can detect particle larger than 0.8 μm in diameter. To make things easier, we'll assume it's capable of that and let the sensor use the PM2.5 device class, reporting in "µg/m³" with 0 accuracy decimals as default.
+
+To use this sensor, you need to setup a voltage sensor source (:doc:`ADC sensor <adc>`, but for example also :doc:`ADS1115 <ads1115>`) and pass it to the sensor with the ``sensor`` option, in order to measure the sensor's output voltage pulses; you also need a binary :ref:`output` to control the sensor's internal IR LED.
```
Comment (optimusprimespace) [inline components/sensor/gp2y1010au0f.rst:14]: I don't know if the ADS1115 is fast enough to capture the rapid signal the sensor is giving off, i have to recheck it, but last time i checked 860SPS is 860Hz ->1.1628ms which is higher than the 0.28ms sampling time required by the sensor. So from the ESPHome catalog of supported ADConverters only the MCP3204/3208, MCP3004/3008 and ADC128S102.

File: components/sensor/gp2y1010au0f.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        name: 'Indoor PM2.5'
+        sensor: adc_sensor
+        adc_voltage_offset: 0.2
+        adc_voltage_multiplier: 3.3
+        output: dust_sensor_led
+        update_interval: 10s
+        filters:
+          - sliding_window_moving_average:
+              window_size: 3
+              send_every: 3
```
Comment (jesserockz) [inline components/sensor/gp2y1010au0f.rst]: Examples should be simpler and specific to this component only
```suggestion
    sensor:
      - platform: gp2y1010au0f
        name: 'Indoor PM2.5'
        sensor: REPLACEME
        output: REPLACEME
```

File: components/sensor/gp2y1010au0f.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+Configuration variables:
+------------------------
+
+- **name** (**Required**, string): The name of the sensor.
+- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
```
Comment (jesserockz) [inline components/sensor/gp2y1010au0f.rst]: We no longer include these on individual pages as they are part of the base sensor schema which is linked and users need to learn.
```suggestion
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by jesserockz — APPROVED** [review]

**Pull request merged**